diff --git a/compiler/erlang_check.erl b/compiler/erlang_check.erl
index e99afa3..9d5004b 100755
--- a/compiler/erlang_check.erl
+++ b/compiler/erlang_check.erl
@@ -203,6 +203,8 @@ disable(Arg) ->
       File :: string().
 check_file(File) ->
     case file_type(File) of
+        hrl ->
+            ok;
         module ->
             check_module(File);
         escript ->
@@ -220,13 +222,18 @@ check_file(File) ->
       File :: string(),
       Result :: module | escript | {error, term()}.
 file_type(File) ->
-    case file:open(File, [raw, read]) of
-        {ok, Fd} ->
-            Result = read_file_type(Fd),
-            ok = file:close(Fd),
-            Result;
-        {error, _Reason}  = Error ->
-            Error
+    case lists:reverse(File) of
+        "lrh" ++ _HrlFile ->
+            hrl;
+        _ ->
+            case file:open(File, [raw, read]) of
+                {ok, Fd} ->
+                    Result = read_file_type(Fd),
+                    ok = file:close(Fd),
+                    Result;
+                {error, _Reason}  = Error ->
+                    Error
+            end
     end.
 
 %%------------------------------------------------------------------------------
@@ -359,15 +366,16 @@ guess_build_system(Path) ->
     % The order is important, at least Makefile needs to come last since a lot
     % of projects include a Makefile along any other build system.
     BuildSystems = [
-                    {rebar3, [
-                              "rebar.lock"
-                             ]
-                    },
                     {rebar, [
+                             "rebar2.config",
                              "rebar.config",
                              "rebar.config.script"
                             ]
                     },
+                    {rebar3, [
+                              "rebar.lock"
+                             ]
+                    },
                     {makefile, [
                             "Makefile"
                            ]
@@ -478,7 +486,7 @@ load_build_files(rebar, _ProjectRoot, ConfigFiles) ->
     load_rebar_files(ConfigFiles, no_config);
 load_build_files(rebar3, ProjectRoot, _ConfigFiles) ->
     % _ConfigFiles is a list containing only rebar.lock.
-    ConfigNames = ["rebar.config", "rebar.config.script"],
+    ConfigNames = ["rebar.config", "rebar2.config", "rebar.config.script"],
     case find_files(ProjectRoot, ConfigNames) of
         [] ->
             log_error("rebar.config not found in ~p~n", [ProjectRoot]),
